ExpenseTracker/operator/operator.go

```diff
@@ -0,0 +1,219 @@
+)
+
+// Describes each expense
+type expense struct {
+	id     int
+	date   time.Time
+	desc   string
+	amount int
+}
+
+// Set of getters
+func (e *expense) Id() int {
+	return e.id
+}
+func (e *expense) Amount() int {
+	return e.amount
+}
+func (e *expense) Date() time.Time {
+	return e.date
+}
+func (e *expense) Desc() string {
+	return e.desc
+}
+
+// If provided data is valid - returns nil, else - returns error
+func ValidateData(description string, amount int) error {
+
+	if amount < 0 {
+		return errors.New("negative amount promoted")
+	}
+	if amount > math.MaxInt {
+		return errors.New("amount is too big")
+	}
+	runes := []rune(description)
+	if len(runes) > 30 { // i do believe that u can describe your purchase in 30 symbols or less
+		return errors.New("your description is too long")
+	}
+
+	return nil
+
+}
+
+func CreateExpense(description string, amount int) expense {
+	return expense{0, time.Now(), description, amount}
+}
+
+// Shows list of available commands
+func Help() {
+	fmt.Println("Доступные команды: 1. add *Description* *amount* -- adds your expense")
+	fmt.Println("Доступные команды: 2. list -- shows list of your expenses")
+	fmt.Println("Доступные команды: 3. summary -- shows amount you spent")
+	fmt.Println("Доступные команды: 4. help -- shows a list of commands")
+	fmt.Println("Доступные команды: 5. delete *id* -- deletes expense with given id(rest of id's stays the same)")
+	fmt.Println("Доступные команды: 6. update *id* *description* *amount* -- updates id if exists")
+	fmt.Println("Доступные команды: 7. exit -- ends a program")
+	fmt.Println("Доступные команды: 8. reset -- resets database")
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// WORKS ONLY WITH LOCAL DB!
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+// Returns a connection to local DB
+func NewConnectionToDB() (*pgx.Conn, error) {
+	conn, err := pgx.Connect(context.Background(), "postgres://postgres:Asdewq232113@localhost:5433/postgres")
+
+	if err != nil {
+		return nil, err
+	}
+	return conn, nil
+}
+
+// Checks that expense if provided id exists
+func ExpenseExists(conn *pgx.Conn, id int) (bool, error) {
+	var exists bool
+	err := conn.QueryRow(context.Background(),
+		`SELECT EXISTS(SELECT 1 FROM public.Expenses WHERE id = $1)`, id).Scan(&exists)
+	return exists, err
+}
+
+// Adds a promoted expense (need to validate)
+func AddExpensToDatabase(conn *pgx.Conn, e expense) error {
+	query := `INSERT INTO public.Expenses (Date, Description, Amount) VALUES (@Date, @Description, @Amount)`
+
+	args := pgx.NamedArgs{
+		"Date":        e.date,
+		"Description": e.desc,
+		"Amount":      e.amount,
+	}
+
+	_, err := conn.Exec(context.Background(), query, args)
+	if err != nil {
+		return fmt.Errorf("error inserting expense: %v", err)

```

Comment:
```suggestion
		return fmt.Errorf("error inserting expense: %w", err)

```
Это нужно, чтобы можно было проверять оригинальную ошибку через `errors.Is`